acceptance-tests/buffer_size_test.go

```diff
@@ -0,0 +1,64 @@
+package acceptance_tests
+
+import (
+	"fmt"
+	"math/rand"
+	"net/http"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gbytes"
+)
+
+var _ = Describe("max_rewrite and buffer_size_bytes", func() {
+	It("Allows HTTP requests as large as buffer_size_bytes - max_rewrite", func() {
+		opsfileBufferSize := `---
+- type: replace
+  path: /instance_groups/name=haproxy/jobs/name=haproxy/properties/ha_proxy/max_rewrite?
+  value: 4096
+- type: replace
+  path: /instance_groups/name=haproxy/jobs/name=haproxy/properties/ha_proxy/buffer_size_bytes?
+  value: 71000
+`
+		haproxyBackendPort := 12000
+		haproxyInfo, _ := deployHAProxy(baseManifestVars{
+			haproxyBackendPort:    haproxyBackendPort,
+			haproxyBackendServers: []string{"127.0.0.1"},
+			deploymentName:        defaultDeploymentName,
+		}, []string{opsfileBufferSize}, map[string]interface{}{}, true)
+
+		closeLocalServer, localPort := startDefaultTestServer()
+		defer closeLocalServer()
+
+		closeTunnel := setupTunnelFromHaproxyToTestServer(haproxyInfo, haproxyBackendPort, localPort)
+		defer closeTunnel()
+
+		By("Sending a request to HAProxy with a 64kb header")
```

Comment:
Can we add another test that expects a 400 if the request exceeds 64k bytes? it should still work at 65k but probably throw an error at 66k, right?

Reply:
66 might still be OK... I'll add one for 72kb which should definitely fail. 

The exact header limit will depend on things like the path, user agent string, and host name so I think it's better to keep a wider gap between what we expect to be allowed and what we expect to be rejected.